### PR TITLE
Avoid warnings in HIP backend

### PR DESCRIPTION
**Pull request by masterleinad**
Marking places that need some TLC with `FIXME` or `FIXME_HIP`.
I don't see any warnings after using `-Wno-unused-command-line-argument -Wno-braced-scalar-init`.

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Space.cpp [hunk 1]
```diff
@@ -529,19 +535,17 @@ void SharedAllocationRecord<Kokkos::Experimental::HIPSpace, void>::
...
 #endif
 }
 
-}  // namespace Impl
-}  // namespace Kokkos
-
 /*--------------------------------------------------------------------------*/
 /*--------------------------------------------------------------------------*/
-namespace Kokkos {
-namespace {
```
Comment (dalg24) [inline core/src/HIP/Kokkos_HIP_Space.cpp]: Why did you remove the anonymous namespace?  What warning was issued for it?

Comment (masterleinad) [reply core/src/HIP/Kokkos_HIP_Space.cpp]: That this function was never used (because it is in an anonymous namespace and not used in that file). For `CUDA` it also lives in the `Kokkos::Impl` namespace and is not used in the same file.

**Comment by Rombur** [comment]
Let's wait to merge this PR. It's going to conflict with a bunch of things I am doing and this is not urgent.

**Comment by Rombur** [comment]
Thanks for waiting...
